pytorch_lightning/core/saving.py

```diff
@@ -361,15 +378,16 @@ def save_hparams_to_yaml(config_yaml, hparams: Union[dict, Namespace]) -> None:
 
     # saving with OmegaConf objects
     if OMEGACONF_AVAILABLE:
-        if OmegaConf.is_config(hparams):
-            with fs.open(config_yaml, "w", encoding="utf-8") as fp:
-                OmegaConf.save(hparams, fp, resolve=True)
-            return
-        for v in hparams.values():
-            if OmegaConf.is_config(v):
-                with fs.open(config_yaml, "w", encoding="utf-8") as fp:
-                    OmegaConf.save(OmegaConf.create(hparams), fp, resolve=True)
+        # deepcopy: hparams from user shouldn't be resolved
```

Comment:
I'm a bit lost as to what the logic here is. Let's say I have multiple DictConfigs stored in hparams like below

```python
{
    'arg1': DictConfig(...),
    'arg2': DictConfig(...)
}
```

With the logic you're proposing, we're going to resolve all DictConfig's eagerly (so that all interpolation's are resolved) and then save the hparams file.

The previous logic would only save the config if it found a DictConfig object, or in my logic the first conf that it finds, which is definitely unsuitable.

So to be clear, the issue is that hparams is a dictionary of DictConfigs and there is no save function that auto resolves them?

Reply:
Just realised the PR got merged, but if we could come up with a solution that doesn't require a try/catch + recursive search this would be preferred I think (cc @tchaton @omry)

Any reason why the try catch was needed?

Reply:
@omry It was auto-merge, but let s come with a better solution.